src/mode/modeHandler.ts

```diff
@@ -1472,6 +1483,47 @@ export class ModeHandler implements vscode.Disposable {
         }
       }
 
+      // Combine instersected selections - When we have multiple cursors
+      // sometimes those cursors selections intersect and combine, we need
+      // to check that here so that we know if our currents cursors will
+      // trigger a selectionChangeEvent or not. If we didn't check for this
+      // vscode might already have the resulting combined selection selected
+      // but since that wouldn't be the same as our selections we would think
+      // there would be a selectionChangeEvent when there wouldn't be any.
+      let combinedSelections: vscode.Selection[] = [];
+      selections.forEach((s, i) => {
+        if (i > 0) {
+          const previousSelection = combinedSelections[i - 1];
+          const overlap = s.intersection(previousSelection);
+          if (overlap) {
+            // If anchor is after active we have a backwards selection and in that case we want
+            // the anchor that is lower and/or to the right and the active that is up and/or to
+            // the left. Otherwise we want the anchor that is upper and/or to the left and the
+            // active that is lower and/or to the right.
+            const anchor = s.anchor.isBeforeOrEqual(s.active)
```

Comment:
I'm not a big fan of nested ternaries - since the first if/else (forward/backward selection) is the same for `anchor` and `active`, can you pull that out into a single real `if`/`else`?

Reply:
I don't like them either, is just that they seem easier to quickly write and test and I always forget to refactor them 😁